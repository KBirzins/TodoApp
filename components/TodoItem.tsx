import React from 'react';
import styled from 'styled-components/native';
import {useDispatch} from 'react-redux';

import CheckBox from './common/CheckBox';

const StyledView = styled.View`
  flex-direction: row;
  padding-horizontal: 30px;
  padding-vertical: 5px;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  background-color: white;
`;

const StyledText = styled.Text`
  margin-left: 5px;
  font-size: 14px;
`;

interface ITodoItem {
  text: string;
  done: boolean;
  timestamp: number;
}

const TodoItem = ({text, done, timestamp}: ITodoItem) => {
  const dispatch = useDispatch();
  return (
    <StyledView>
      <CheckBox
        selected={done}
        onPress={() => dispatch({type: 'TOGGLE_TODO', timestamp})}
      />
      <StyledText>{text}</StyledText>
    </StyledView>
  );
};

export default TodoItem;
