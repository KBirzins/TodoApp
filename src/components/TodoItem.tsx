import React, {useState} from 'react';
import styled from 'styled-components/native';
import {useDispatch} from 'react-redux';
import {TouchableWithoutFeedback} from 'react-native';

import CheckBox from './common/CheckBox';
import Icon from './common/Icon';

const StyledView = styled.View`
  flex-direction: row;
  padding-horizontal: 30px;
  padding-vertical: 5px;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
`;

const StyledText = styled.Text<{done: boolean}>`
  margin-left: 5px;
  font-size: 14px;
  text-decoration-line: ${props => (props.done ? 'line-through' : 'none')};
`;

const StyledTextInput = styled.TextInput`
  margin-left: 5px;
  font-size: 14px;
  width: 80%;
  padding-vertical: 0px;
  padding-horizontal: 0px;
`;

const ContainerView = styled.View`
  flex-direction: row;
  justify-content: space-between;
  width: 100%;
  padding-right: 30px;
`;

const ButtonContainerView = styled.View`
  flex-direction: row;
`;

interface ITodoItem {
  text: string;
  done: boolean;
  timestamp: number;
  isEditing?: boolean;
  onEditingDone(): any;
}

const TodoItem = ({
  text,
  done,
  timestamp,
  isEditing,
  onEditingDone,
}: ITodoItem) => {
  const dispatch = useDispatch();
  const [todoText, setText] = useState(text);

  return (
    <TouchableWithoutFeedback
      onPress={() => {
        dispatch({type: 'TOGGLE_TODO', timestamp});
      }}>
      <StyledView>
        <CheckBox
          selected={done}
          onPress={() => {
            dispatch({type: 'TOGGLE_TODO', timestamp});
          }}
        />
        {isEditing ? (
          <>
            <ContainerView>
              <StyledTextInput
                editable={isEditing}
                onChangeText={text => setText(text)}
                value={todoText}
                autoFocus
              />
              <ButtonContainerView>
                <Icon
                  name="check"
                  onPress={() => {
                    onEditingDone();
                    dispatch({type: 'EDIT_TODO', timestamp, text: todoText});
                  }}
                />
                <Icon
                  name="close"
                  onPress={() => {
                    setText(text);
                    onEditingDone();
                  }}
                />
              </ButtonContainerView>
            </ContainerView>
          </>
        ) : (
          <StyledText done={done}>{text}</StyledText>
        )}
      </StyledView>
    </TouchableWithoutFeedback>
  );
};

export default TodoItem;
