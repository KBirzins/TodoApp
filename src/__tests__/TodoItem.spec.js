import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TodoItem from '../components/TodoItem';
import {ReduxProvider} from '../tests/helpers';

describe('TodoItem', () => {
  it('renders correctly with correct properties', () => {
    const component = renderer.create(
      <ReduxProvider>
        <TodoItem
          text="Some text"
          done={true}
          timestamp={1579199193297}
          isEditing={false}
        />
      </ReduxProvider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
