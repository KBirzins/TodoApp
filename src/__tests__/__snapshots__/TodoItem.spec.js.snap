// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders correctly with correct properties 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingHorizontal": 30,
        "paddingVertical": 5,
        "width": "100%",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "color": "#211f30",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î ´
  </Text>
  <Text
    done={true}
    style={
      Array [
        Object {
          "fontSize": 14,
          "marginLeft": 5,
          "textDecorationLine": "line-through",
        },
      ]
    }
  >
    Some text
  </Text>
</View>
`;
